app/serializers/suggested_topics_mixin.rb

```diff
@@ -16,6 +17,25 @@ def include_suggested_topics?
     object.next_page.nil? && object.suggested_topics&.topics
   end
 
+  def include_suggested_group_name?
+    return false unless include_suggested_topics?
+    object.topic.private_message? && scope.user
+  end
+
+  def suggested_group_name
+    return nil if object.topic.topic_allowed_users.exists?(user_id: scope.user.id)
+
+    if object.topic_allowed_group_ids.present?
+      Group
+        .joins(:group_users)
+        .where(
+          "group_users.group_id IN (?) AND group_users.user_id = ?",
+          object.topic_allowed_group_ids, scope.user
```

Comment:
Should be `scope.user.id` I think?

Reply:
O wow it should be but ActiveRecord apparently has some magic to use the user_id when composing the SQL. No wonder my tests worked.